From 6b5b773488a4ab77b5c18d82e663e68692f14b33 Mon Sep 17 00:00:00 2001
From: Ashod Nakashian <ashod.nakashian@collabora.co.uk>
Date: Thu, 13 Oct 2016 15:52:32 -0400
Subject: tdf#103185 - Add VCL backend being used to About dialog

The VCL backend is now shown on the About dialog.

Change-Id: Ib4e6b33e9c2b3ec2fca229fed8f4533227a67ac6
Reviewed-on: https://gerrit.libreoffice.org/29798
Reviewed-by: Yousuf Philips <philipz85@hotmail.com>
Tested-by: Yousuf Philips <philipz85@hotmail.com>
Tested-by: Jenkins <ci@libreoffice.org>
Reviewed-by: Michael Meeks <michael.meeks@collabora.com>

diff --git a/vcl/inc/svids.hrc b/vcl/inc/svids.hrc
index efb3490..891fe57 100644
--- a/vcl/inc/svids.hrc
+++ b/vcl/inc/svids.hrc
@@ -179,6 +179,7 @@
 #define SV_APP_UIRENDER                             10802
 #define SV_APP_GL                                   10803
 #define SV_APP_DEFAULT                              10804
+#define SV_APP_VCLBACKEND                           10805
 
 #define SV_ICON_SIZE48_START                        20000
 #define SV_ICON_SIZE32_START                        21000
diff --git a/vcl/source/app/svapp.cxx b/vcl/source/app/svapp.cxx
index 33e6fe8..61e0072 100644
--- a/vcl/source/app/svapp.cxx
+++ b/vcl/source/app/svapp.cxx
@@ -1211,6 +1211,13 @@ OUString Application::GetHWOSConfInfo()
         aDetails.append( VclResId(SV_APP_DEFAULT).toString() );
     aDetails.append( "; " );
 
+#ifdef LINUX
+    // Only linux has different backends, so don't show blank for others.
+    aDetails.append( VclResId(SV_APP_VCLBACKEND).toString() );
+    aDetails.append( GetToolkitName() );
+    aDetails.append( "; " );
+#endif
+
     return aDetails.makeStringAndClear();
 }
 
diff --git a/vcl/source/src/app.src b/vcl/source/src/app.src
index 91d9df1..b36686d 100644
--- a/vcl/source/src/app.src
+++ b/vcl/source/src/app.src
@@ -34,4 +34,9 @@ String SV_APP_DEFAULT
     Text [ en-US ] = "default";
 };
 
+String SV_APP_VCLBACKEND
+{
+    Text [ en-US ] = "VCL: ";
+};
+
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
-- 
cgit v0.10.2

